src/OpenLoco/Input/ShortcutManager.cpp

```diff
@@ -299,13 +299,41 @@ namespace OpenLoco::Input::ShortcutManager
         window->invalidate();
     }
 
+    static void toggleWindowCore(WindowType windowType, std::function<void()> windowOpener)
+    {
+        auto window = WindowManager::find(windowType);
+
+        // window isn't open
+        if (window == nullptr)
+        {
+            windowOpener();
+            return;
+        }
+
+        // same window, so close it or bring to front
+        if (WindowManager::isInFront(window))
+        {
+            WindowManager::close(window);
+        }
+        else
+        {
+            WindowManager::bringToFront(window);
+        }
+    }
+
+    template<typename TFunc, typename... TFuncArgs>
+    static void toggleWindow(WindowType windowType, TFunc windowOpener, TFuncArgs... args)
```

Comment:
helper method so we don't have to call std::bind at every call site

Reply:
Couldn't you just write lambdas instead of messing with a bind?

Reply:
Yes, but that would have been more code. I can do the lambdas if you prefer, but I think its much cleaner this way